ui/components/app/whats-new-popup/whats-new-popup.js

```diff
@@ -212,15 +263,29 @@ const renderSubsequentNotification = ({
 export default function WhatsNewPopup({ onClose }) {
   const t = useContext(I18nContext);
   const history = useHistory();
+  const dispatch = useDispatch();
 
   const notifications = useSelector(getSortedAnnouncementsToShow);
+  const smartTransactionsOptInStatus = useSelector(
+    getSmartTransactionsOptInStatus,
+  );
   const locale = useSelector(getCurrentLocale);
 
   ///: BEGIN:ONLY_INCLUDE_IF(blockaid)
   const theme = useTheme();
   ///: END:ONLY_INCLUDE_IF
 
+  const [stxOptInSent, setStxOptInSent] = useState(false);
   const [seenNotifications, setSeenNotifications] = useState({});
+
+  const stxNotification = notifications.find(
+    (notification) => notification.id === NOTIFICATION_SMART_TRANSACTIONS,
+  );
+  if (stxNotification && !smartTransactionsOptInStatus && !stxOptInSent) {
```

Comment:
We needed to trigger some custom logic when the STX notification is displayed to a user. If there is some other way to do it, let me know.